tests/test.sh

```diff
@@ -32,7 +32,7 @@ expect() {
 
     echo "$expected_output" > "$tmp_expected"
 
-    "$fd" "$@" | sort -f > "$tmp_output"
+    "$fd" "$@" | xargs -0 echo -n | sort -f > "$tmp_output"
```

Comment:
Instead of modifying `expect()` here, couldn't we somehow check for the string `"result1\0result2\0result3"` below?